src/utilities/ApiGatewayAuth.js

```
@@ -1,29 +1,39 @@
 import axios from 'axios';
+import qs from 'qs';
 
 export class ApiGatewayAuth {
   constructor(config = {}) {
     this.url = config.url || process.env.API_GATEWAY_AUTH_URL;
     this.clientId = config.clientId || process.env.API_GATEWAY_AUTH_CLIENT_ID;
     this.clientSecret =
       config.clientSecret || process.env.API_GATEWAY_AUTH_CLIENT_SECRET;
-    this.token = config.token;
-    // TODO: Configurable?
-    this.grantType = 'CLIENT_CREDENTIALS';
+    this.token = config.token || false;
+    this.grantType = 'client_credentials';
   }
 
   isAuthenticated = () => !!this.token;
 
   authenticate = async () => {
-    const params = {
+    const body = {
       client_id: this.clientId,
       client_secret: this.clientSecret,
       grant_type: this.grantType,
     };
-    const result = await axios.post(this.url, undefined, {
-      params,
-    });
-    this.token = result.data;
-
-    return this.token;
+    await axios
+      .post(this.url, qs.stringify(body))
+      .then((response) => {
+        this.token = response.data;
+        return this.token;
+      })
+      .catch((error) => {
```

Err. I added some conditional console.log()s for Axios error handling but is there a more elegant way to present these errors? 🤔

---

Logging the error to the console is good but you should also throw the error so that the MegaProfile client aborts the call to the API as you are not authenticated to do so.